subsys/net/lib/coap/coap.c

```diff
@@ -1397,3 +1399,17 @@ struct coap_observer *coap_find_observer_by_addr(
 
 	return NULL;
 }
+
+u16_t coap_next_id(void)
+{
+	static u16_t message_id;
+	static atomic_t seeded;
+
+	/* According to CoAP specification, message id should be initialized to
+	 * a random number, not to zero. */
+	if (atomic_cas(&seeded, 0, 1)) {
```

Comment:
This certainly works but there is still possibility that one CoAP message would have message id 1. For example if the `sys_rand32_get()` would block then it could cause some other thread to run, then that other thread could possible get message_id = 1 as a result. Chance for this is very minimal.

One solution for that could be that we introduce `coap_init()` function that could be called from `services_init()` which is in `subsys/net/ip/net_core.c`, and which could do the message_id init.

Reply:
Agree that's way more idiomatic

Reply:
I interpreted the "standard" approach to the best of my ability, but feel free to place some extra attention on `net_core.c`. There doesn't seem to be a consensus best practice for conditionally NOPing functions so I just sort of picked one.